config/interpolate_funcs.go

```diff
@@ -95,6 +96,19 @@ func interpolationFuncFile() ast.Function {
 	}
 }
 
+// interpolationFuncBase64 implements the "base64" function to
+// encode a string in base64
+func interpolationFuncBase64() ast.Function {
+	return ast.Function{
+		ArgTypes:     []ast.Type{ast.TypeString},
+		ReturnType:   ast.TypeString,
+		Callback: func(args []interface{}) (interface{}, error) {
+			toEncode := []byte(args[0])
```

Comment:
``args[0]`` here is of type ``interface{}`` rather than ``[]byte``, so this won't compile.

You can think of an interface in Go as a pointer to both a value and a type, allowing the type to vary at runtime. In order to access the underlying value you must use a *type assertion*, where you tell Go which type you believe should be present and it will return the underlying value if you're right and panic (crash) if not. So adding a type assertion here as you wrote it would look like this:

In `ArgTypes` above you've stated that this argument is a string, so the caller of this function (the interpolation code) will guarantee that the interface is wrapping a ``string`` here, so we can just assert that it's a string and know it will never fail as long as there are no bugs in the caller. So you can rewrite this line as follows:

```go
    toEncode: []byte(args[0].(string))
```

This `.(type)` syntax is the type assertion, which simply retrieves the string that the interface value points at. This is distinct from a type *conversion*, which is what you were already doing to convert the string to ``[]byte``; in this case the Go runtime copies the characters from the string to a new byte slice, discarding the original string value and leaving it for the garbage collector to clean up.